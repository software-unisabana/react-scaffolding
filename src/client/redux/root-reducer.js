import { combineReducers } from 'redux';

import itemReducer from './item/item.reducer';

const rootReducer = combineReducers({
	products: itemReducer,
});

export default rootReducer;
