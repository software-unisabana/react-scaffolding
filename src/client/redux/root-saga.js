import { all, call } from 'redux-saga/effects';

import { itemSagas } from './item/item.sagas';

export default function* rootSaga() {
	yield all([call(itemSagas)]);
}
