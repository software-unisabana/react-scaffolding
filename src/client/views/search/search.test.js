import React from 'react';
import { mount } from 'enzyme';
import { combineReducers, createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import SearchView from './search.component';

export const createMockStore = ({ state, reducers }) => {
	const store = createStore(combineReducers(reducers), state);
	return {
		...store,
	};
};

describe('SearchView', () => {
	let wrapper;
	let mockFetchItemsStart;
	let store;

	beforeEach(() => {
		const mockReducer = (
			state = {
				isFetching: true,
			},
			action
		) => state;

		const mockState = {
			products: {
				isFetching: true,
			},
		};

		mockFetchItemsStart = jest.fn();

		store = createMockStore({
			state: mockState,
			reducers: { products: mockReducer },
		});

		const mockLocation = {
			search: 'ipod',
		};

		const mockProps = {
			location: mockLocation,
			fetchItemsStart: mockFetchItemsStart,
		};

		wrapper = mount(
			<BrowserRouter>
				<Provider store={store}>
					<SearchView {...mockProps} />
				</Provider>
			</BrowserRouter>
		);
	});
	it('should render SearchView component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
