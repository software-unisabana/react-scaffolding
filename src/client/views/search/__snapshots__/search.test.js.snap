// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView should render SearchView component 1`] = `ReactWrapper {}`;
